src/components/QuranReader/TranslationView/TranslationText/index.tsx

```diff
@@ -19,7 +19,7 @@ import Footnote from 'types/Footnote';
 interface Props {
   translationFontScale: number;
   text: string;
-  resourceName: string;
+  resourceName?: string;
```

Comment:
IMO, this should be left as it was and we introduce a new `boolean` called `showResourceName` or something similar.

Reply:
I have a few reason to keep the API this way. In my opinion we should keep the API Surface Area minimal and a component should only receive the data it needs.

an implementation in a `Card` component that only need to show `title`, looks like this

```
<Card title="card title" />
```

vs

```
<Card title="card title" description={null} coverImage={null} shouldShowDescription={false} shouldShowCoverImage={false} />
```

the first example is better IMO

Reply:
Yes I agree. Thanks Muhajir!